### PR TITLE
Clarify semantics of auto traits.

### LINKED ISSUE #257: Documentation for auto traits makes incorrect claims about Sync.
https://doc.rust-lang.org/nightly/reference/special-types-and-traits.html#auto-traits

This page claims that if `T` is `Sync`, then so is `&mut T`. This is false, however. I believe that it's false for `RefUnwind` as well.

### PR DESCRIPTION
**Pull request by alercah**
Fixes #257.

### CODE REVIEW

File: src/special-types-and-traits.md [hunk 1]
```diff
@@ -96,12 +98,23 @@ First, auto traits are automatically implemented using the following rules:
   closure that captures a `T` by shared reference and a `U` by value implements
   any auto traits that both `&T` and `U` do.
 
+For generic types (counting the built-in types above as generic over `T`), if an
+generic implementation is available, then the compiler does not automatically
+implement it for types that could use the implementation except that they do not
+meet the requisite trait bounds. For instance, the standard library implements
+`Send` for all `&T` where `T` is `Sync`; this means that the compiler will not
+implement `Send` for `&T` for if `T` is `Send` but not `Sync`.
```
Comment (Havvy) [inline src/special-types-and-traits.md]: "for if"?

File: src/special-types-and-traits.md [hunk 2]
```diff
@@ -96,12 +98,23 @@ First, auto traits are automatically implemented using the following rules:
...
 implementations. For example `*mut T` has a negative implementation of `Send`,
-and so `*mut T` and `(*mut T,)` are not `Send`. Finally, auto traits may
-be added as a bound to any [trait object]\: `Box<Debug + Send + UnwindSafe>` is
-a valid type.
+and so `*mut T` is not `Send`, even if `T` is. There is currently no stable way
+to specify additional negative implementations; they exist only in the standard
+library.
+
+Auto traits may be added as an additional bound to any [trait object], even
+though normally only one trait is allowed. For instance, `Box<Debug + Send +
```
Comment (Havvy) [inline src/special-types-and-traits.md]: Wanna add a `dyn` here?

**Comment by alercah** [comment]
Fixed issues.

**Comment by Havvy** [comment]
💟 Thanks!

**Pull request merged**